### ISSUE TITLE
`jinjalint-parse-error` on valid HTML

### ISSUE THREAD

**Issue by dlukeomalley** [returntocorp/bento#280]
## Overview
`r2c.jinja` reports parsing errors on https://github.com/nickjj/build-a-saas-app-with-flask

## Current Behavior

```bash
$ bento check --tool=r2c.jinja --all
r2c.jinja has not been configured. Adding default configuration for tool to config.yml
Running Bento checks on all tracked files...

r2c.jinja․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․|■■■■■■■■■■| 00:05, 🍱 Done

r2c.jinja jinjalint-parse-error
     > assets/static/502.html:30
     = Parse error: expected one of '[:a-z]', 'area', 'base', 'br', 'col',
       'command', 'embed', 'hr', 'img', 'input', 'keygen', 'link', 'meta',
       'param', 'script', 'source', 'style', 'track', 'wbr', '{#', '{%', '{{' at
       30:1

r2c.jinja jinjalint-parse-error
     > assets/static/maintenance.html:30
     = Parse error: expected one of '[:a-z]', 'area', 'base', 'br', 'col',
       'command', 'embed', 'hr', 'img', 'input', 'keygen', 'link', 'meta',
       'param', 'script', 'source', 'style', 'track', 'wbr', '{#', '{%', '{{' at
       30:1

r2c.jinja jinjalint-missing-meta-content-type
     > snakeeyes/templates/layouts/base.html:1
     = HTML missing a meta content-type declaration may result in content
       misinterpretation in certain browsers, especially when using the
       'file://' protocol, and thus XSS. Include a meta content-type like '<meta
       http-equiv="Content-Type" content="text/html; charset=UTF-8">' to avoid
       misinterpretation.

r2c.jinja jinjalint-parse-error
     > snakeeyes/templates/macros/form.html:39
     = Parse error: expected one of 'autoescape', 'block', 'blocktrans',
       'comment', 'filter', 'for', 'if', 'ifchanged', 'ifequal', 'ifnotequal',
       'not an intermediate Jinja tag name', 'spaceless', 'verbatim', 'with' at
       39:5

⚠ 4 finding(s) in this project in 5.64 s

Please fix these issues, or:
```

Opening these files, they appear to be valid HTML:

![image](https://user-images.githubusercontent.com/605906/74793029-03b37d00-5274-11ea-9e2e-ef8db56367ae.png)

```html
<!DOCTYPE html>
<head>
  <title>Reload Your Browser</title>
  <style>
    body {
      background-color: #fff;
      width: 960px;
      margin: 0 auto;
      font-family: Arial, Helvetica, sans-serif;
    }
    h1 {
      font-size: 42px;
      color: #000;
      padding-bottom: 5px;
      border-bottom: 1px solid #000;
    }
    p {
      color: #444;
      font-size: 22px;
    }
  </style>
</head>
<body>
  <h1>We Just Updated Our Site</h1>
  <h2>When we release a new version of our site it takes a second to come back up.</h2>
  <p>It should be fixed by the time you read this message. Try reloading your browser.</p>
  <p><i>P.S., don't worry, your data is safely backed up!</i></p>
  <hr>
  <p>Seeing this page for a while? Please let us know by emailing <a href="mailto:nick.janetakis@gmail.com">nick.janetakis@gmail.com</a>.</p>
</body>
</html>
```

## Steps to Reproduce

1. `$ git clone git@github.com:nickjj/build-a-saas-app-with-flask.git`
2. `$ bento check --tool=r2c.jinja --all`

## Additional Context
Bento version (run `bento --version`): 0.9.1

**Comment by dlukeomalley** [comment]
CC @mschwager @minusworld

**Comment by minusworld** [comment]
Looking at the example listed for assets/static/502.html, it _technically_ isn't valid HTML because it is missing the opening <html> tag.  The same is true for assets/static/maintenance.html.  Adding these in fixes the parse error.

Unfortunately, the parse error message is totally unhelpful.  I've made an issue on jinjalint to address this: https://github.com/returntocorp/jinjalint/issues/5

Still investigating snakeeyes/templates/macros/form.html.

**Comment by minusworld** [comment]
For snakeeyes/templates/macros/form.html, I've run it through two different validators (jinja-ninja and straight-up rendering the template using the jinja engine) and it seems to be fine.

It seems that jinjalint's parsing is a bit lacking.  We could investigate jinjalint's parsing engine... but we are considering adding html template parsing to sgrep (https://github.com/returntocorp/sgrep/issues/182), so we should re-implement the jinja checks in sgrep once support is added.

To summarize this issue, returntocorp/jinjalint#5 will be addressed but the parsing issues themselves won't be addressed unless it's deemed absolutely necessary.

**Comment by dlukeomalley** [comment]
I have been seeing more and more `ToolRunExceptions` in the logs, and I'm wondering if `r2c.jinja` is the culprit (per PRIVACY.md, I can't see what tool failed).

Have we run it at an appreciable scale and does it have a low failure rate?

**Comment by minusworld** [comment]
This run here (https://dev.massive.ret2.co/jobs/405?page=1&sort_on=repo_url&sort_by=asc&error_type=AnalyzerNonZeroExitError) shows 51 failures out of 1,199 repos, which is about 4%.

That was also before https://github.com/returntocorp/jinjalint/issues/3 was fixed, which was the cause of 7 out of 10 errors I sampled from that run.

(That leads to a new approx. error rate of 1.3%)

**Comment by dlukeomalley** [comment]
@minusworld Reflecting on this more, how many of the 1,199 projects have either HTML files or Jinja templates? Based on "in the wild" error rates I'm seeing I'm worried the input set isn't set up to properly vet the analyzer.

**Comment by nbrahms** [comment]
Closing as wont-fix. Instead we are disabling the `parse-error` rule by default.
